File: src/main/java/com/bm/transfer/transaction/controller/TransactionController.java
package com.bm.transfer.transaction.controller;

import com.bm.transfer.transaction.dto.TransactionPageResponse;
import com.bm.transfer.transaction.dto.request.TransactionRequestHistory;
import com.bm.transfer.transaction.dto.response.TransactionResponseDto;
import com.bm.transfer.transaction.entity.Transaction;
import com.bm.transfer.transaction.service.TransactionService;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/transactions")
@RequiredArgsConstructor
@CrossOrigin(
        origins = {
                "http://localhost:4200"
        }
)
public class TransactionController {

    private final TransactionService service;


    @ResponseBody
    @GetMapping("/transaction")
    public TransactionPageResponse getTransactionsHistory(
            @RequestParam(defaultValue = "0") int pageNo,
            @RequestParam(defaultValue = "5") int pageSize,
            @RequestParam(defaultValue = "id") String sortBy,
            @RequestParam("account-number") String accountNumber
    ){
        return  service.getTransactionsHistory(pageNo, pageSize, sortBy, accountNumber);
    }
}

File: src/main/java/com/bm/transfer/controller/TransactionController.kt
package com.bm.transfer.controller

import com.bm.transfer.dto.response.TransactionPageResponse
import com.bm.transfer.service.TransactionService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.responses.ApiResponse
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/transactions")
@Tag(name = "Transactions", description = "Endpoints for managing transaction history")
class TransactionController(
    private val service: TransactionService
) {

    @CrossOrigin(origins = ["*"])
    @Operation(
        summary = "Get transaction history",
        description = "Retrieves the transaction history for a specific account",
        responses = [
            ApiResponse(responseCode = "200", description = "Transaction history retrieved successfully"),
            ApiResponse(responseCode = "400", description = "Invalid request parameters"),
            ApiResponse(responseCode = "404", description = "Account not found")
        ]
    )
    @GetMapping("/transaction")
    fun getTransactionsHistory(
        @RequestParam(defaultValue = "0") pageNo: Int,
        @RequestParam(defaultValue = "5") pageSize: Int,
        @RequestParam(defaultValue = "id") sortBy: String,
        @RequestParam("account-number") accountNumber: String
    ): TransactionPageResponse {
        return service.getTransactionsHistory(pageNo, pageSize, sortBy, accountNumber)
    }
}